contracts/src/Gateway.sol

```diff
@@ -617,4 +621,31 @@ contract Gateway is IGateway, IInitializable {
         assets.assetHubCreateAssetFee = config.assetHubCreateAssetFee;
         assets.assetHubReserveTransferFee = config.assetHubReserveTransferFee;
     }
+
+    /// @inheritdoc IGateway
+    function transact(
+        ParaID destinationChain,
+        OriginKind originKind,
+        uint128 destinationFee,
+        Weight calldata weightAtMost,
+        bytes calldata call
+    ) external payable {
+        if (call.length < 2 || destinationFee == 0 || weightAtMost.refTime == 0 || weightAtMost.proofSize == 0) {
+            revert InvalidTransact();
+        }
+        Ticket memory ticket;
+        Costs memory costs;
+        bytes memory payload =
+            SubstrateTypes.Transact(msg.sender, originKind.encode(), destinationFee, weightAtMost, call);
+        ticket.dest = destinationChain;
+        ticket.costs = costs;
+        ticket.payload = payload;
```

Comment:
I would rather initialise the ticket like this to make clear that we aren't mistakenly leaving costs uninitialized
```suggestion
Ticket memory ticket = Ticket({
  dest: destinationChain,
  costs: Costs({
    native: 0,
    foreign: 0,
  }),
  payload: payload
});
```